docs/source/getting-started.md

```diff
@@ -90,20 +90,35 @@ Please follow the links below to learn specific details about how to enable/conf
 
-**Important requirements regarding the nodes**
+#### Important requirements regarding the nodes
 
 We have two cases:
 
-1. The kernel is run in (Docker) container: in that case, the image should ensure the availability of the kernel libraries and kernelspec. The kernelspec is not necessary here, only the launcher. We talk about this in container customization.
-2. The kernelspecs are only required on all nodes if using the DistributedProcessProxy - which can apply to YARN Client mode, Standalone, and 
-Distributed. All kernels (libraries...) and their corresponding kernelspecs must reside on each nodes. The kernelspec hierarchies (i.e., paths) must be available and identical on all nodes. This is not applicable to Apache Toree in Yarn cluster mode since spark-submit will transfer the files at startup. SSH passwordless is needed between the EG node and the other nodes.
+*Case 1 - The kernel is run in (Docker) container.*
+
+In that case, the image should ensure the availability of the kernel libraries and kernelspec. The kernelspec is not necessary here, only the launcher. We talk about this in [container customization](./docker.html#bringing-your-own-kernel-image).
+
+The launch of containerized kernels (at least via EG) is two-fold. First, there's the argv section in the kernelspec that is processed by EG. In these cases, the command that is invoked is a python script using the container's api (kubernetes, docker, or docker swarm) that is responsible for converting any necessary "parameters" to environment variables, etc. that are used during the actual container creation.
+
+The command that is then run in the container is THEN the actual kernel launcher script. This launcher is responsible for taking the response address (which is now an env variable) and returning the kernel's connection information back on that response address to EG. The kernel launcher does additional things - but primarily listens for interrupt and shutdown requests, which it then passes along to the actual (embedded) kernel.
+
+So container envs have two launches - one to launch the container itself, the other to launch the kernel (within the container).
+
+*Case 2 - The kernel is run via DistributedProcessProxy*
+
+The kernelspecs are only required on all nodes if using the DistributedProcessProxy - which apply to YARN Client mode, Standalone, and 
+Distributed modes. All kernels (libraries...) and their corresponding kernelspecs must reside on each nodes. The kernelspec hierarchies (i.e., paths) must be available and identical on all nodes. This is not applicable to Apache Toree in Yarn cluster mode since spark-submit will transfer the files at startup. SSH passwordless is needed between the EG node and the other nodes.
```

Comment:
hmm.  I think we could use a third case here since theres some convolution going on.  How about something like this...

*Case 2 - The kernel is run via DistributedProcessProxy*

The kernelspecs are required on all nodes if using the `DistributedProcessProxy` - which apply to YARN Client mode, Standalone, and Distributed modes. All kernels (libraries...) and their corresponding kernelspecs must reside on each node. The kernelspec hierarchies (i.e., paths) must be available and identical on all nodes.  SSH passwordless is needed between the EG node and the other nodes.

*Case 3 - The kernel is run via YarnClusterProcessProxy or ConductionClusterProcessProxy*

With _cluster_ process proxies, distribution of kernelspecs to all nodes besides the EG node is not required. However, the IPython and R Kernel kernel must be installed on each node. Note that because the Apache Toree kernel, and its supporting libraries, will be transferred to the target node via spark-submit, installation of Apache Toree (the scala kernel) is not required except on the Enterprise Gateway node itself.


Based on this, I'm thinking Case 1 may want to read something like...

*Case 1 - The kernel is run in via a container-based process proxy (Kubernetes, Docker or DockerSwarm)*

Not sure the parenthesized details are necessary but I think the use of 'process proxy' is more consistent across the 3.